### PR TITLE
[SIW-1171] Add Function App env variables

### PR DESCRIPTION
**Pull request by silvicir**
#### List of Changes

1. Added `nonces` cosmos container with TTL
2. Added function app environment variables

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/resources/_modules/function_apps/locals.tf [hunk 1]
```diff
@@ -13,8 +13,33 @@ locals {
       FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
       FETCH_KEEPALIVE_TIMEOUT             = "60000"
 
-      // CosmosDB connection
-      COSMOS_DB_ENDPOINT = var.cosmos_db_endpoint
-    }
+      CosmosDbConnectionString = "AccountEndpoint=${var.cosmos_db_endpoint};AccountKey=${var.cosmos_db_key};"
+      CosmosDbDatabaseName     = var.cosmos_database_names[0]
+
+      FederationEntityBasePath         = "https://io-d-wallet-it.azurewebsites.net"
```
Comment (grausof) [inline infra/resources/_modules/function_apps/locals.tf:19]: The CND URL should be placed here

File: infra/resources/_modules/function_apps/locals.tf [hunk 2]
```diff
@@ -13,8 +13,33 @@ locals {
...
+      FederationEntityOrganizationName = "PagoPa S.p.A."
+      FederationEntityHomepageUri      = "https://io.italia.it"
+      FederationEntityPolicyUri        = "https://io.italia.it/privacy-policy/"
+      FederationEntityTosUri           = "https://io.italia.it/privacy-policy/"
```
Comment (grausof) [inline infra/resources/_modules/function_apps/locals.tf:23]: Ok for now. When we have the privacy policy they must be changed

File: infra/resources/_modules/function_apps/locals.tf [hunk 3]
```diff
@@ -13,8 +13,33 @@ locals {
...
+      FederationEntityLogoUri          = "https://io.italia.it/assets/img/io-it-logo-blue.svg"
+      FederationEntityTrustAnchorUri   = "https://demo.federation.eudi.wallet.developers.italia.it"
+      IosBundleIdentifier              = "it.pagopa.app.io"
+      IosTeamIdentifier                = "DSEVY6MV9G"
+      AndroidBundleIdentifier          = "it.pagopa.app.io"
+      AndroidCrlUrl                    = "https://android.googleapis.com/attestation/status"
+      AndroidPlayIntegrityUrl          = "https://www.googleapis.com/auth/playintegrity"
+      AndroidPlayStoreCertificateHash  = "INVALID"
```
Comment (grausof) [inline infra/resources/_modules/function_apps/locals.tf:31]: @LazyAfternoons Can you give us the correct value?

File: infra/resources/_modules/function_apps/locals.tf [hunk 4]
```diff
@@ -13,8 +13,33 @@ locals {
...
+      GooglePublicKey                  = "-----BEGIN PUBLIC KEY-----\nMIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAr7bHgiuxpwHsK7Qui8xU\nFmOr75gvMsd/dTEDDJdSSxtf6An7xyqpRR90PL2abxM1dEqlXnf2tqw1Ne4Xwl5j\nlRfdnJLmN0pTy/4lj4/7tv0Sk3iiKkypnEUtR6WfMgH0QZfKHM1+di+y9TFRtv6y\n//0rb+T+W8a9nsNL/ggjnar86461qO0rOs2cXjp3kOG1FEJ5MVmFmBGtnrKpa73X\npXyTqRxB/M0n1n/W9nGqC4FSYa04T6N5RIZGBN2z2MT5IKGbFlbC8UrW0DxW7AYI\nmQQcHtGl/m00QLVWutHQoVJYnFPlXTcHYvASLu+RhhsbDmxMgJJ0mcDpvsC4PjvB\n+TxywElgS70vE0XmLD+OJtvsBslHZvPBKCOdT0MS+tgSOIfga+z1Z1g7+DVagf7q\nuvmag8jfPioyKvxnK/EgsTUVi2ghzq8wm27ud/mIM7AY2qEORR8Go3TVB4HzWQgp\nZrt3i5MIlCaY504LzSRiigHCzAPlHws+W0rB5N+er5/2pJKnfBSDiCiFAVtCLOZ7\ngLiMm0jhO2B6tUXHI/+MRPjy02i59lINMRRev56GKtcd9qO/0kUJWdZTdA2XoS82\nixPvZtXQpUpuL12ab+9EaDK8Z4RHJYYfCT3Q5vNAXaiWQ+8PTWm2QgBR/bkwSWc+\nNpUFgNPN9PvQi8WEg5UmAGMCAwEAAQ==\n-----END PUBLIC KEY-----"
+      AllowDevelopmentEnvironment      = "true"
```
Comment (grausof) [inline infra/resources/_modules/function_apps/locals.tf:34]:
```suggestion
      AllowDevelopmentEnvironment      = "false"
```

**Review by grausof — CHANGES_REQUESTED** [review]

File: infra/resources/_modules/function_apps/locals.tf (hunk 3 above)
Comment (LazyAfternoons) [reply infra/resources/_modules/function_apps/locals.tf:31]: The value for `io-app`, extracted directly from the APK and matched on Google Play Console is: 
`7D:E4:FE:3E:AA:E0:83:FF:CD:8B:07:03:01:E8:65:02:D3:F1:EA:D1:DD:66:AC:14:2B:AD:7C:54:47:55:4F:82`

image

Comment (grausof) [reply infra/resources/_modules/function_apps/locals.tf]: I think it should be converted first in HEX or base64 string. @LazyAfternoons can you check how it comes from the attestation?

Comment (LazyAfternoons) [reply infra/resources/_modules/function_apps/locals.tf]: According to our [DR](https://pagopa.atlassian.net/wiki/spaces/SIW/pages/921436266/IO-WALLET-DR-0007+Play+Integrity+API) and the [official documentation](https://developer.android.com/google/play/integrity/verdicts#application-integrity-field), the SHA-256 is an URL-safe base64 encoded string: 
```json
appIntegrity: {
  // PLAY_RECOGNIZED, UNRECOGNIZED_VERSION, or UNEVALUATED.
  appRecognitionVerdict: "PLAY_RECOGNIZED"
  // The package name of the app.
  // This field is populated iff appRecognitionVerdict != UNEVALUATED.
  packageName: "com.package.name"
  // The sha256 digest of app certificates (base64-encoded URL-safe).
  // This field is populated iff appRecognitionVerdict != UNEVALUATED.
  certificateSha256Digest: ["6a6a1474b5cbbb2b1aa57e0bc3"]
  // The version of the app.
  // This field is populated iff appRecognitionVerdict != UNEVALUATED.
  versionCode: "42"
}
```

Comment (grausof) [reply infra/resources/_modules/function_apps/locals.tf]: Ok so it should be: `feT-Pqrgg__NiwcDAehlAtPx6tHdZqwUK618VEdVT4I`

**Review by grausof — APPROVED** [review]

**Pull request merged**